src/docmanager/index.ts

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IContentsModel, IKernelId, IKernelSpecId, IContentsOpts, IKernel
+} from 'jupyter-js-services';
+
+import {
+  IDisposable, DisposableDelegate
+} from 'phosphor-disposable';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+
+/**
+ * The interface for a document model.
+ */
+export interface IDocumentModel {
```

Comment:
I suppose this could be `any`, since a notebook model could return JSON and a text model would return a string.